schematic/utils/general.py

```diff
@@ -26,3 +27,30 @@ def unlist(_list):
         return _list[0]
     else:
         return _list
+
+def get_dir_size(path='.'):
+    '''
+    calculate total size of a directory
+    args: 
+    path: path to a folder or directory
+    '''
+    total = 0
+    with os.scandir(path) as it:
+        for entry in it:
+            if entry.is_file():
+                total += entry.stat().st_size
+            elif entry.is_dir():
+                total += get_dir_size(entry.path)
+    return total
+
+def convert_size(size_bytes):
+    '''
+    convert bytes to a human readable format
+    '''
+    if size_bytes == 0:
+       return "0B"
+    size_name = ("B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB")
+    i = int(math.floor(math.log(size_bytes, 1024)))
```

Comment:
Nit: `i` is not a descriptive variable.  I would move away from single letter variables for code readability.